hazelcast/src/main/java/com/hazelcast/internal/eviction/EvictionPolicyEvaluatorProvider.java

```diff
@@ -43,27 +44,27 @@ private static EvictionPolicyComparator createEvictionPolicyComparator(EvictionP
     }
 
     /**
-     * Gets the {@link EvictionPolicyEvaluator} implementation specified with <code>evictionPolicy</code>.
+     * Gets the {@link EvictionPolicyEvaluator} implementation specified with {@code evictionPolicy}.
      *
      * @param evictionConfig {@link EvictionConfiguration} for requested {@link EvictionPolicyEvaluator} implementation
      * @param classLoader    the {@link java.lang.ClassLoader} to be used
      *                       while creating custom {@link EvictionPolicyComparator} if it is specified in the config
      * @return the requested {@link EvictionPolicyEvaluator} implementation
      */
-    public static EvictionPolicyEvaluator getEvictionPolicyEvaluator(EvictionConfiguration evictionConfig,
-                                                                     ClassLoader classLoader) {
+    public static <A, E extends Evictable> EvictionPolicyEvaluator<A, E> getEvictionPolicyEvaluator(
+            EvictionConfiguration evictionConfig, ClassLoader classLoader) {
         if (evictionConfig == null) {
             return null;
         }
 
-        EvictionPolicyComparator evictionPolicyComparator = null;
+        EvictionPolicyComparator evictionPolicyComparator;
 
         String evictionPolicyComparatorClassName = evictionConfig.getComparatorClassName();
         if (!StringUtil.isNullOrEmpty(evictionPolicyComparatorClassName)) {
             try {
                 evictionPolicyComparator = ClassLoaderUtil.newInstance(classLoader, evictionPolicyComparatorClassName);
             } catch (Exception e) {
-                ExceptionUtil.rethrow(e);
+                throw rethrow(e);
```

Comment:
I'm not 100% with this extra `throw`, but due to the `rethrow` signature IntelliJ won't stop complaining and cannot see that `evictionPolicyComparator` will always be set if no exception is thrown.